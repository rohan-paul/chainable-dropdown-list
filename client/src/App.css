html,body {
  margin: 0;
  padding: 0;
  height: 100vh;
  width: 100vw;
}
.container {
  height: 100vh;
  width: 100vw;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}
.filter-form {
  height: 1000px;
  width: 1000px;
  border: solid 1px gold;
  overflow: auto;
}
.filter-item {
  background: gold;
  height: 30px;
  padding: 5px 10px;
  color: black;
  border-bottom: solid 1px goldenrod;
}
.filter-options {

  display: flex;
  margin-bottom: 20px;
}
.filter-option {
  margin: 0 10px;
}
label, select {
  margin-right: 10px;
}