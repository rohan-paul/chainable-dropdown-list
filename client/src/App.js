import React, { Component } from 'react';
import './App.css';
import axios from 'axios';
// import Navbar from './components/layout/Navbar'

import FilterOptions from './components/FilterOptions';
import FilterItems from './components/FilterItems';

class App extends React.Component {

    constructor(props) {
      super(props)

      this.state = {
        data: [],
        carMake: '',
        model: '',
        multiple: true
      }
    }

    componentDidMount(){
       axios.get('/api')
      .then(response => {
        const data = response.data
        this.setState({ data });
      })
      .catch(function (error) {
        console.log(error);
      })
    }

  filterItems = (val, type) => {
     switch (type) {
      case 'carMake':
        this.setState({carMake: val});
        break;
      case 'model':
        this.setState({model: val});
        break;
      default:
        break;
    }
  }

  render () {

    let filteredItems = this.state.data;
    let state = this.state;
    let filterProperties = ["carMake", "model"];

    filterProperties.forEach(filterBy => {
      let filterValue = state[filterBy];

      if (filterValue) {
        filteredItems = filteredItems.filter(item =>item[filterBy] === filterValue);
      }
    });

    let carMakeArray = this.state.data.map(item => item.carMake );
    let modelArray = this.state.data.map(item => item.model );

    // In the drop-down add an extra empty row at the beginning.
    carMakeArray.unshift("");
    modelArray.unshift("");


    return (
      <div className="container">
        <FilterOptions

            data={this.state.data}
            carMakeOptions={carMakeArray}
            modelOptions={modelArray}
            changeOptionProp={this.filterItems} />

        <div className="filter-form">
          <FilterItems key={filteredItems.name} data={filteredItems} />
        </div>
      </div>
    )
  }
};

export default App;

/* The changeOptionProp above is an example of how data is being passed from parent to child - https://medium.com/@ruthmpardee/passing-data-between-react-components-103ad82ebd17

Within the parent component, I have the 'filterItems' variable which has access to the data. But I need my child component to have access too. So, I pass it as a prop to the child.
  */