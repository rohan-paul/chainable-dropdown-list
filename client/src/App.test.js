import React from 'react';
import ReactDOM from 'react-dom';
// import App from '../tests/App';
import App from './App';

import { shallow, mount } from 'enzyme';


/*it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});*/

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('shallow render without crashing', () => {
  const wrapper = shallow(<App />);
  expect(wrapper).toMatchSnapshot();
});

/*it('should change the state after change the input value', () => {
  const newValue = 'testing component';
  const wrapper = mount(<App />);
  const input = wrapper.find('input');

  input.simulate('change', { target: { value: newValue }});

  expect(wrapper.state().basicValue).toEqual(newValue);
});*/