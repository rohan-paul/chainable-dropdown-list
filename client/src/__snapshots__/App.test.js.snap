// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow render without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <FilterOptions
          carMakeOptions={
                    Array [
                              "",
                            ]
          }
          changeOption={[Function]}
          data={Array []}
          modelOptions={
                    Array [
                              "",
                            ]
          }
/>,
        <div
          className="filter-form"
>
          <FilterItems
                    data={Array []}
          />
</div>,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "carMakeOptions": Array [
            "",
          ],
          "changeOption": [Function],
          "data": Array [],
          "modelOptions": Array [
            "",
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <FilterItems
            data={Array []}
/>,
          "className": "filter-form",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "data": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <FilterOptions
            carMakeOptions={
                        Array [
                                    "",
                                  ]
            }
            changeOption={[Function]}
            data={Array []}
            modelOptions={
                        Array [
                                    "",
                                  ]
            }
/>,
          <div
            className="filter-form"
>
            <FilterItems
                        data={Array []}
            />
</div>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "carMakeOptions": Array [
              "",
            ],
            "changeOption": [Function],
            "data": Array [],
            "modelOptions": Array [
              "",
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <FilterItems
              data={Array []}
/>,
            "className": "filter-form",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "data": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
